src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -95,64 +99,80 @@ const usePushNotificationHook = () => {
       anonymousToken
     });
   };
+
   let isOpenNotification = false;
 
-  const helperNavigationResetWithData = (screenData) => {
+  const baseResetNavigation = (tabsScreenName: HomeTabsScreen, screen: ScreenData) => {
     setTimeout(() => {
-      if (!isOpenNotification) {
-        isOpenNotification = true;
-        const routes = [
-          {
-            name: 'AuthenticatedStack',
+      const routes = [
+        {
+          name: 'AuthenticatedStack',
+          params: {
+            screen: 'HomeTabs',
             params: {
-              screen: 'HomeTabs',
-              params: {
-                screen: 'SignedChannelList'
-              }
+              screen: tabsScreenName
             }
           }
-        ];
-
-        if (screenData) {
-          routes.push({
-            name: 'AuthenticatedStack',
-            params: {
-              ...screenData
-            }
-          });
+        },
+        {
+          name: 'AuthenticatedStack',
+          params: {
+            ...screen
+          }
         }
+      ];
 
-        navigation.reset({
-          index: screenData ? 2 : 1,
-          routes
-        });
-      }
+      navigation.reset({
+        index: screen?.screen ? 2 : 1,
+        routes
+      });
     }, 500);
   };
 
+  const helperNavigationResetWithData = (screenData) => {
+    if (!isOpenNotification) {
+      isOpenNotification = true;
+      baseResetNavigation('SignedChannelList', {
+        ...screenData
+      });
+    }
+  };
+
   const __handleNotification = async (notification) => {
     if (notification?.data?.type === 'topic' && notification?.data?.topic_name) {
+      AnalyticsEventTracking.eventTrack(
+        `${BetterSocialEventTracking.BACKEND_PUSH_NOTIFICATIONS}topicPageScreen`
+      );
       navigation.navigate('TopicPageScreen', {
         id: notification?.data?.topic_name
       });
     }
 
     if (notification?.data?.type === 'feed' || notification?.data?.type === 'reaction') {
+      AnalyticsEventTracking.eventTrack(
+        `${BetterSocialEventTracking.BACKEND_PUSH_NOTIFICATIONS}postDetailPage`
+      );
       navigation.navigate('PostDetailPage', {
         feedId: notification.data.feed_id,
         is_from_pn: true
       });
     }
     if (notification?.data?.type === 'follow_user') {
+      AnalyticsEventTracking.eventTrack(
+        `${BetterSocialEventTracking.BACKEND_PUSH_NOTIFICATIONS}followedUser`
+      );
       navigation.navigate('OtherProfile', {
         data: {
-          user_id: notification?.data?.user_id,
+          user_id: signedProfileId,
           other_id: notification?.data?.user_id_follower,
           username: notification?.data?.username_follower
         }
       });
     }
     if (notification?.data?.type === 'message.new') {
+      AnalyticsEventTracking.eventTrack(
+        `${BetterSocialEventTracking.BACKEND_PUSH_NOTIFICATIONS}newMessage`
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are mostly good, but there's a potential issue with the `isOpenNotification` variable. It's defined as a local variable in the hook and is used to prevent multiple notifications from being processed at once. However, due to the nature of hooks and how they're re-rendered, this variable will be reset to `false` on every render, which could lead to unexpected behavior if multiple notifications arrive at the same time.

To fix this, you could use a `useRef` hook to persist the `isOpenNotification` value across renders:

```diff
- let isOpenNotification = false;
+ const isOpenNotification = useRef(false);

// Then replace all instances of isOpenNotification with isOpenNotification.current
```
This way, the value of `isOpenNotification` will persist across re-renders and maintain its expected functionality.


<!-- This is an auto-generated comment by OSS CodeRabbit -->